In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('E:/Datasets/target_action_prediction_SberAvtopodpiska/ga_hits.csv')
df.head(20)

session_id    hit_date   hit_time  \
0   5639623078712724064.1640254056.1640254056  2021-12-23   597864.0   
1   7750352294969115059.1640271109.1640271109  2021-12-23   597331.0   
2    885342191847998240.1640235807.1640235807  2021-12-23   796252.0   
3    142526202120934167.1640211014.1640211014  2021-12-23   934292.0   
4   3450086108837475701.1640265078.1640265078  2021-12-23   768741.0   
5   6466333295973247896.1640246168.1640246168  2021-12-23   704194.0   
6    281215510786615563.1640245516.1640245516  2021-12-23   911854.0   
7   4024492994895054107.1640269084.1640269084  2021-12-23  1071804.0   
8    555009234841130092.1640256620.1640256620  2021-12-23  1013899.0   
9   2692901778487480807.1640206845.1640206845  2021-12-23        0.0   
10  1945627655676404094.1640276355.1640276355  2021-12-23      197.0   
11  4555345648396008371.1640233907.1640233907  2021-12-23      107.0   
12  5636135307570360848.1640280592.1640280592  2021-12-23       12.0   
13  6993187768713388524.1640276467.1640276467  2021-12-23     1186.0   
14  9063469723165129646.1640233902.1640233902  2021-12-23       68.0   
15  9221311837172398962.1640233841.1640233841  2021-12-23      506.0   
16  8279923128365523547.1640276571.1640276571  2021-12-23     1024.0   
17  9126144797936188300.1640233869.1640233869  2021-12-23      488.0   
18  1333511261702627793.1640276433.1640276433  2021-12-23     5618.0   
19   699466478407410560.1640289549.1640289549  2021-12-23       20.0   

    hit_number hit_type           hit_referer  \
0           30    event                   NaN   
1           41    event                   NaN   
2           49    event                   NaN   
3           46    event                   NaN   
4           79    event                   NaN   
5           41    event                   NaN   
6          102    event                   NaN   
7           85    event                   NaN   
8          101    event  VloVXNWduHeTjUoDkjkO   
9            1    event                   NaN   
10           4    event                   NaN   
11           4    event  trXmoxcphNjuYcShOXwL   
12           3    event                   NaN   
13           4    event                   NaN   
14           3    event  uMnMdxLOIznakHyXLoMC   
15           3    event  trXmoxcphNjuYcShOXwL   
16           3    event  uMnMdxLOIznakHyXLoMC   
17           3    event                   NaN   
18           6    event                   NaN   
19           6    event                   NaN   

                                        hit_page_path event_category  \
0   sberauto.com/cars?utm_source_initial=google&ut...           quiz   
1   sberauto.com/cars/fiat?city=1&city=18&rental_c...           quiz   
2   sberauto.com/cars/all/volkswagen/polo/e994838f...           quiz   
3   sberauto.com/cars?utm_source_initial=yandex&ut...           quiz   
4   sberauto.com/cars/all/mercedes-benz/cla-klasse...           quiz   
5   sberauto.com/cars?utm_source_initial=youtube&u...           quiz   
6   sberauto.com/cars/toyota?isnew=false&rental_ca...           quiz   
7   sberauto.com/cars/all/mercedes-benz/glc/f8f330...           quiz   
8   sberauto.com/cars/all/kia/sorento/c38179cb?utm...           quiz   
9   sberauto.com/cars/all/nissan/x-trail/0744675f?...       card_web   
10  sberauto.com/cars/all/porsche/911/f149c91b?ren...       card_web   
11  sberauto.com/cars/all/nissan/qashqai/bfc21661?...       card_web   
12  sberauto.com/cars/all/renault/duster/145d53f1?...       card_web   
13  sberauto.com/cars/all/porsche/911/f149c91b?ren...       card_web   
14  sberauto.com/cars/all/skoda/rapid/bf24b977?ren...       card_web   
15  sberauto.com/cars/all/skoda/rapid/bf24b977?ren...       card_web   
16  sberauto.com/cars/all/porsche/911/f149c91b?ren...       card_web   
17  sberauto.com/cars/all/renault/logan/8c3c73f2?r...       card_web   
18  sberauto.com/cars/all/porsche/911/f149c91b?ren...       card_web   
19  sberauto.com/cars/all/peugeot/traveller/4c1dd8..

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15726470 entries, 0 to 15726469
Data columns (total 11 columns):
 #   Column          Dtype  
---  ------          -----  
 0   session_id      object 
 1   hit_date        object 
 2   hit_time        float64
 3   hit_number      int64  
 4   hit_type        object 
 5   hit_referer     object 
 6   hit_page_path   object 
 7   event_category  object 
 8   event_action    object 
 9   event_label     object 
 10  event_value     float64
dtypes: float64(2), int64(1), object(8)
memory usage: 1.3+ GB


In [4]:
df.describe(include='all')

session_id    hit_date      hit_time  \
count                                    15726470    15726470  6.566148e+06   
unique                                    1734610         226           NaN   
top     5442565791571325612.1632449195.1632449195  2021-05-24           NaN   
freq                                          768      513035           NaN   
mean                                          NaN         NaN  2.091050e+05   
std                                           NaN         NaN  4.032110e+05   
min                                           NaN         NaN  0.000000e+00   
25%                                           NaN         NaN  2.412000e+04   
50%                                           NaN         NaN  8.994100e+04   
75%                                           NaN         NaN  2.195352e+05   
max                                           NaN         NaN  1.315688e+07   

          hit_number  hit_type           hit_referer           hit_page_path  \
count   1.572647e+07  15726470               9451666                15726470   
unique           NaN         1                 37873                  342715   
top              NaN     event  HbolMJUevblAbkHClEQa  podpiska.sberauto.com/   
freq             NaN  15726470               8879187                 2793639   
mean    2.356715e+01       NaN                   NaN                     NaN   
std     2.887713e+01       NaN                   NaN                     NaN   
min     1.000000e+00       NaN                   NaN                     NaN   
25%     7.000000e+00       NaN                   NaN                     NaN   
50%     1.500000e+01       NaN                   NaN                     NaN   
75%     2.900000e+01       NaN                   NaN                     NaN   
max     5.000000e+02       NaN                   NaN                     NaN   

       event_category event_action           event_label  event_value  
count        15726470     15726470              11966286          0.0  
unique             52          230                 39825          NaN  
top          card_web    view_card  KclpemfoHstknWHFiLit          NaN  
freq          7456998      3558985               6505447          NaN  
mean              NaN          NaN                   NaN          NaN  
std               NaN          NaN                   NaN          NaN  
min               NaN          NaN                   NaN          NaN  
25%               NaN          NaN                   NaN          NaN  
50%               NaN          NaN                   NaN          NaN  
75%               NaN          NaN                   NaN          NaN  
max               NaN          NaN                   NaN          NaN

In [5]:
df2 = pd.read_csv('E:/Datasets/target_action_prediction_SberAvtopodpiska/ga_sessions.csv', low_memory=False)
df2.head(20)

session_id              client_id  \
0   9055434745589932991.1637753792.1637753792  2108382700.1637753791   
1    905544597018549464.1636867290.1636867290   210838531.1636867288   
2   9055446045651783499.1640648526.1640648526  2108385331.1640648523   
3   9055447046360770272.1622255328.1622255328  2108385564.1622255328   
4   9055447046360770272.1622255345.1622255345  2108385564.1622255328   
5   9055447192389856083.1622453074.1622453074  2108385598.1622453075   
6   9055455318486370642.1640843788.1640843788  2108387490.1640843602   
7   9055461992850812764.1626107740.1626107740  2108389044.1626107740   
8   9055462349345527315.1638536723.1638536723  2108389127.1638536723   
9   9055466554104774132.1624800757.1624800757  2108390106.1624800756   
10  9055466554104774132.1629442326.1629442326  2108390106.1624800756   
11  9055469620715506713.1628883994.1628883994  2108390820.1628883993   
12  9055469620715506713.1633110583.1633110583  2108390820.1628883993   
13  9055469620715506713.1635878177.1635878177  2108390820.1628883993   
14  9055469620715506713.1636559415.1636559415  2108390820.1628883993   
15  9055476398179250925.1634235280.1634235280  2108392398.1634235117   
16  9055487268745225369.1637983385.1637983385  2108394929.1637983385   
17  9055487955932503299.1630493885.1630493885  2108395089.1630493955   
18  9055488359664698205.1635763037.1635763037  2108395183.1635763037   
19  9055491722626447422.1638119489.1638119489  2108395966.1638119486   

    visit_date visit_time  visit_number            utm_source utm_medium  \
0   2021-11-24   14:36:32             1  ZpYIoDJMcFzVoPFsHGJL     banner   
1   2021-11-14   08:21:30             1  MvfHsxITijuriZxsqZqt        cpm   
2   2021-12-28   02:42:06             1  ZpYIoDJMcFzVoPFsHGJL     banner   
3   2021-05-29   05:00:00             1  kjsLglQLzykiRbcDiGcD        cpc   
4   2021-05-29   05:00:00             2  kjsLglQLzykiRbcDiGcD        cpc   
5   2021-05-31   12:00:00             1  kjsLglQLzykiRbcDiGcD    organic   
6   2021-12-30   08:56:28             1  TxKUcPpthBDPieTGmVhx        cpc   
7   2021-07-12   19:00:00             1  ZpYIoDJMcFzVoPFsHGJL     banner   
8   2021-12-03   16:05:23             1  ZpYIoDJMcFzVoPFsHGJL     banner   
9   2021-06-27   16:00:00             1  ZpYIoDJMcFzVoPFsHGJL     banner   
10  2021-08-20   09:00:00             2  ZpYIoDJMcFzVoPFsHGJL     banner   
11  2021-08-13   22:00:00             1  fDLlAcSmythWSCVMvqvL     (none)   
12  2021-10-01   20:00:00             2  fDLlAcSmythWSCVMvqvL     (none)   
13  2021-11-02   21:36:17             3  gVRrcxiDQubJiljoTbGm   referral   
14  2021-11-10   18:50:15             4  SzZERoLMmrEUEhDaYcyN        cpc   
15  2021-10-14   21:00:00             1  ZpYIoDJMcFzVoPFsHGJL     banner   
16  2021-11-27   06:23:05             1  ZpYIoDJMcFzVoPFsHGJL     banner   
17  2021-09-01   13:00:00             1  ZpYIoDJMcFzVoPFsHGJL     banner   
18  2021-11-01   13:37:17             1  ZpYIoDJMcFzVoPFsHGJL     banner   
19  2021-11-28   20:11:29             1  ZpYIoDJMcFzVoPFsHGJL     banner   

            utm_campaign         utm_adcontent           utm_keyword  \
0   LEoPHuyFvzoNfnzGgfcd  vCIpmpaGBnIQhyYNkXqp  puhZPIYqKXeFPaUviSjo   
1   FTjNLDyTrXaWYgZymFkV  xhoenQgDQsgfEPYNPwKO  IGUCNvHlhfHpROGclCit   
2   LEoPHuyFvzoNfnzGgfcd  vCIpmpaGBnIQhyYNkXqp  puhZPIYqKXeFPaUviSjo   
3                    NaN  NOBKLgtuvqYWkXQHeYWM                   NaN   
4                    NaN                   NaN                   NaN   
5   LTuZkdKfxRGVceoWkVyg  JNHcPlZPxEMWDnRiyoBf                   NaN   
6   FTjNLDyTrXaWYgZymFkV  LcGIUNPUAmXtQJaDfFBR  PwscUHjoUJDrtfWESIHj   
7   LEoPHuyFvzoNfnzGgfcd  JNHcPlZPxEMWDnRiyoBf                   NaN   
8   LEoPHuyFvzoNfnzGgfcd  vCIpmpaGBnIQhyYNkXqp  puhZPIYqKXeFPaUviSjo   
9   LEoPHuyFvzoNfnzGgfcd  JNHcPlZPxEMWDnRiyoBf                   NaN   
10  LEoPHuyFvzoNfnzGgfcd  JNHcPlZPxEMWDnRiyoBf                   NaN   
11  LTuZkdKfxRGVceoWkVyg  JNHcPlZPxEMWDnRiyoBf               

In [6]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1860042 entries, 0 to 1860041
Data columns (total 18 columns):
 #   Column                    Dtype 
---  ------                    ----- 
 0   session_id                object
 1   client_id                 object
 2   visit_date                object
 3   visit_time                object
 4   visit_number              int64 
 5   utm_source                object
 6   utm_medium                object
 7   utm_campaign              object
 8   utm_adcontent             object
 9   utm_keyword               object
 10  device_category           object
 11  device_os                 object
 12  device_brand              object
 13  device_model              object
 14  device_screen_resolution  object
 15  device_browser            object
 16  geo_country               object
 17  geo_city                  object
dtypes: int64(1), object(17)
memory usage: 255.4+ MB


In [7]:
df2.describe(include='all')

session_id              client_id  \
count                                     1860042                1860042   
unique                                    1860042                1391719   
top     5162860564725728638.1634488716.1634488716  1750498477.1629450062   
freq                                            1                    462   
mean                                          NaN                    NaN   
std                                           NaN                    NaN   
min                                           NaN                    NaN   
25%                                           NaN                    NaN   
50%                                           NaN                    NaN   
75%                                           NaN                    NaN   
max                                           NaN                    NaN   

        visit_date visit_time  visit_number            utm_source utm_medium  \
count      1860042    1860042  1.860042e+06               1859945    1860042   
unique         226      85318           NaN                   293         56   
top     2021-05-24   12:00:00           NaN  ZpYIoDJMcFzVoPFsHGJL     banner   
freq         39453      61067           NaN                578290     552272   
mean           NaN        NaN  2.712804e+00                   NaN        NaN   
std            NaN        NaN  1.182907e+01                   NaN        NaN   
min            NaN        NaN  1.000000e+00                   NaN        NaN   
25%            NaN        NaN  1.000000e+00                   NaN        NaN   
50%            NaN        NaN  1.000000e+00                   NaN        NaN   
75%            NaN        NaN  2.000000e+00                   NaN        NaN   
max            NaN        NaN  5.640000e+02                   NaN        NaN   

                utm_campaign         utm_adcontent           utm_keyword  \
count                1640439               1524427                777981   
unique                   412                   286                  1219   
top     LTuZkdKfxRGVceoWkVyg  JNHcPlZPxEMWDnRiyoBf  puhZPIYqKXeFPaUviSjo   
freq                  463481               1006599                506819   
mean                     NaN                   NaN                   NaN   
std                      NaN                   NaN                   NaN   
min                      NaN                   NaN                   NaN   
25%                      NaN                   NaN                   NaN   
50%                      NaN                   NaN                   NaN   
75%                      NaN                   NaN                   NaN   
max                      NaN                   NaN                   NaN   

       device_category device_os device_brand          device_model  \
count          1860042    789904      1492864                 16338   
unique               3        13          205                   104   
top             mobile   Android        Apple  AuMdmADEIoPXiWpTsBEj   
freq           1474871    464054       551088                  9778   
mean               NaN       NaN          NaN                   NaN   
std                NaN       NaN          NaN                   NaN   
min                NaN       NaN          NaN                   NaN   
25%                NaN       NaN          NaN                   NaN   
50%                NaN       NaN          NaN                   NaN   
75%                NaN       NaN          NaN                   NaN   
max                NaN       NaN          NaN                   NaN   

       device_screen_resolution device_browser geo_country geo_city  
count                   1860042        1860042     1860042  1860042  
unique                     5039             57         166     2548  
top                     414x896         Chrome      Russia   Moscow  
freq                     169090        1013436     1800565   805329  
mean                        NaN            NaN   

In [8]:
df.duplicated().any()

False

In [9]:
df2.duplicated().any()

False

In [10]:
df['session_id'].unique()

array(['5639623078712724064.1640254056.1640254056',
       '7750352294969115059.1640271109.1640271109',
       '885342191847998240.1640235807.1640235807', ...,
       '3349670872968620291.1640264206.1640264206',
       '1009509786569589790.1640244938.1640244938',
       '5381267721977892188.1640271196.1640271196'], dtype=object)

In [11]:
df[df.event_action.isin(['sub_car_claim_click', 'sub_car_claim_submit_click', 'sub_open_dialog_click',
                         'sub_custom_question_submit_click', 'sub_call_number_click',
                         'sub_callback_submit_click', 'sub_submit_success', 'sub_car_request_submit_click'])]

session_id    hit_date  hit_time  \
4016      2744563715298057088.1640258436.1640258436  2021-12-23  843092.0   
4045      3087297479839089634.1640268774.1640268774  2021-12-23  194144.0   
4046      3156966333326004302.1640206419.1640206800  2021-12-23  327223.0   
4047      3750243879753098158.1640272208.1640272208  2021-12-23  156992.0   
4048      7518333712042258254.1640258901.1640258901  2021-12-23  170616.0   
...                                             ...         ...       ...   
15725025  1277864870843199549.1636773954.1636773954  2021-11-13  354017.0   
15725133   965861352491898977.1636751459.1636751459  2021-11-13  172590.0   
15725134  4727705172767122620.1636818847.1636818847  2021-11-13  356120.0   
15725135  4727705172767122620.1636818847.1636818847  2021-11-13  312303.0   
15725136   116562085450617659.1636832061.1636832061  2021-11-13  124047.0   

          hit_number hit_type hit_referer  \
4016              81    event         NaN   
4045              22    event         NaN   
4046              63    event         NaN   
4047              20    event         NaN   
4048              16    event         NaN   
...              ...      ...         ...   
15725025          30    event         NaN   
15725133          18    event         NaN   
15725134          43    event         NaN   
15725135          41    event         NaN   
15725136          26    event         NaN   

                                              hit_page_path    event_category  \
4016      sberauto.com/cars/all/kia/rio/fee33fe6?utm_sou...        sub_submit   
4045      sberauto.com/cars/all/skoda/rapid/bf24b977?utm...        sub_submit   
4046      sberauto.com/cars/all/skoda/rapid/bf24b977?utm...        sub_submit   
4047      sberauto.com/cars/all/nissan/x-trail/0744675f?...        sub_submit   
4048      sberauto.com/cars/all/mercedes-benz/gla-klasse...        sub_submit   
...                                                     ...               ...   
15725025  sberauto.com/cars/all/lada-vaz/vesta/2fc745ed?...  sub_button_click   
15725133  sberauto.com/cars/all/volkswagen/polo/e994838f...        sub_submit   
15725134  sberauto.com/cars/all/volkswagen/polo/e994838f...        sub_submit   
15725135  sberauto.com/cars/all/volkswagen/polo/e994838f...  sub_button_click   
15725136  sberauto.com/cars/all/volkswagen/polo/e994838f...  sub_button_click   

                        event_action           event_label  event_value  
4016              sub_submit_success  nsPPIRqjxBefONGPpnsF          NaN  
4045              sub_submit_success  nsPPIRqjxBefONGPpnsF          NaN  
4046              sub_submit_success  nsPPIRqjxBefONGPpnsF          NaN  
4047              sub_submit_success  nsPPIRqjxBefONGPpnsF          NaN  
4048              sub_submit_success  KuMiABMMbspIDDhiCNVS          NaN  
...                              ...                   ...          ...  
15725025  sub_car_claim_submit_click                   NaN          NaN  
15725133          sub_submit_success  nsPPIRqjxBefONGPpnsF          NaN  
15725134          sub_submit_success  uimgZZmhfLQwbKAZZfCk          NaN  
15725135       sub_open_dialog_click  ZaZuwAXOKlbzyhUqtnmk          NaN  
15725136  sub_car_claim_submit_click                   NaN          NaN  

[104908 rows x 11 columns]

In [12]:
df2[df2['session_id'] == '9055794766915104430.1624305327.1624305327']

session_id              client_id  \
78  9055794766915104430.1624305327.1624305327  2108466524.1624305326   

    visit_date visit_time  visit_number            utm_source utm_medium  \
78  2021-06-21   22:00:00             1  BHcvLfOaCWvWTykYqHVe        cpc   

   utm_campaign utm_adcontent utm_keyword device_category device_os  \
78          NaN           NaN         NaN         desktop       NaN   

   device_brand device_model device_screen_resolution device_browser  \
78          NaN          NaN                 834x1112         Safari   

   geo_country    geo_city  
78      Russia  Domodedovo

In [13]:
df[df['session_id'] == '9055794766915104430.1624305327.1624305327']

session_id    hit_date  hit_time  \
5593195  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5594864  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5594950  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5595467  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5605713  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5605736  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5606138  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5606173  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5621553  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5621559  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5621575  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5621883  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5621906  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5621910  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5622967  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5622979  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5623008  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5623408  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5623447  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5623608  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5623928  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5623995  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5623999  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5624160  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5624181  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5624221  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5624288  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5624299  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5624322  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5624455  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5624481  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5624483  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5626638  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5626654  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5626701  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5628464  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5628611  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5628878  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5630262  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5631623  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5631625  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5631888  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5631896  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5632516  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5633912  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5634067  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5634435  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5637684  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5637685  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5644414  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   
5644415  9055794766915104430.1624305327.1624305327  2021-06-21       NaN   

         hit_number hit_type           hit_referer  \
5593195           2    event  Hbo

In [14]:
df.isna().any()

session_id        False
hit_date          False
hit_time           True
hit_number        False
hit_type          False
hit_referer        True
hit_page_path     False
event_category    False
event_action      False
event_label        True
event_value        True
dtype: bool

In [15]:
df2.isna().any()

session_id                  False
client_id                   False
visit_date                  False
visit_time                  False
visit_number                False
utm_source                   True
utm_medium                  False
utm_campaign                 True
utm_adcontent                True
utm_keyword                  True
device_category             False
device_os                    True
device_brand                 True
device_model                 True
device_screen_resolution    False
device_browser              False
geo_country                 False
geo_city                    False
dtype: bool

# 1. EDA.

1.1. Датасет представлен двумя таблицами. В обеих таблицах отсутствуют дубли.

1.2. Исходя из количества уникальных строк сессий в обеих таблицах и общего количества строк хитов становится очевидно, что существуют сессии, как без единого хита, так и сессии где хитов 1 и более.

1.3. Идентификатор сессии нужен для объединения таблиц, в качестве фичи в нем нет смысла, поэтому после объединения фича идет на удаление.

1.4. Количество уникальных клиентов говорит о том, что в период, который охватывает датасет существуют клиенты, которые посещали сайт не один раз. В уникальном идентификаторе клиента также нет смысловой нагрузки. Фича на удаление.

1.5. Также анализ данных показал, что в датасете отсутствуют представления о визитах начиная с первого для каждого из клиентов, соответственно невозможно учесть посещение клиентов с аккумулированием времени визита. Таким образом для модели будет считаться, что каждый визит уникален и независим друг от друга.

1.6. В дате визита отсутствуют пропуски.

1.7. Во времени визита отсутствуют пропуски и это время указывает нам на время начала сессии.

1.8. В порядковом номере визита отсутствуют пропуски, однако как уже было отмечено выше, нет цельного представления о каждом визите клиента начиная с первого.

1.9. Информация о канале привлечения содержит 97 пропусков. При 293 уникальных значений имеет смысл замены на наиболее часто встречающийся или наиболее часто встречающийся применительно к типу привлечения.

1.10. В типе привлечения отсутствуют пропуски.

1.11. Информация о рекламной компании содержит 219603 пропуска, что составляет около 12% от общего количества. Удаление этой фичи не видится целесообразным, однако например для значения  utm_source = BHcvLfOaCWvWTykYqHVe, что составляет около половины всех пропусков какие-либо данные отсутствуют, поэтому в рекламной компании все пропуски будем заменять на значение other.

1.12. Информация utm_adcontent содержит 335615 пропусков, что составляет около 18% от общего количества. Удаление этой фичи не видится целесообразным, однако например для значения  utm_source = BHcvLfOaCWvWTykYqHVe, что составляет около трети всех пропусков какие-либо данные отсутствуют, поэтому в utm_adcontent все пропуски будем заменять на значение other.

1.13. Информация utm_keyword содержит пропусков более 50% от общего количества. Фича на удаление.

1.14. Информация device_category не содержит пропусков.

1.15. Информация device_os содержит пропусков более 50% от общего количества. Фича на удаление.

1.16. Информация device_brand содержит 367178 пропусков, что составляет около 20% от общего количества. Удаление этой фичи не видится целесообразным, поэтому в device_brand все пропуски будем заменять на значение other.

1.17. Информация device_model подавляющее количество пропусков. Фича на удаление.

1.18. Информация device_screen_resolution, device_browser, geo_country, geo_city не содержат пропусков.

1.19. Информация hit_number, hit_page_path, event_category, event_action не содержат пропусков.

1.20. Информация hit_date не содержит пропусков, однако более 50% количество пропусков в hit_time делает нецелесообразным использование данного поля, при том, что время начала сессии все равно известно и находится в visit_time. Комплексно продолжительность сессии здесь будет учитываться за счет времени начала сессии и как количества так и порядкового номера нажатия в hit_number. Фича на удаление. 

1.21. Информация hit_time содержит пропусков более 50% от общего количества. Фича на удаление.

1.22. Информация hit_type не содержит пропусков, однако значения представлены единственным уникальным. Фича на удаление.

1.23. Информация hit_referer содержит 6274804 пропусков, что составляет около 40% от общего количества. Фича на удаление.

1.24. Информация hit_page_path представляет строку запроса. Наиболее ценное, что можно извлечь это brand и model автомобиля на странице которого произошло событие. Однако это не всегда возможно, поскольку событие происходило не карточке автомобиля, а где-то еще, поэтому в остальных случаях будем принимать значение none.

1.25. Информация event_category не является независимой и лишь выполняет группировки для фич event_action, которая для нас является целевой. Фича на удаление.

1.26. Информация event_label содержит 3760184 пропусков, что составляет около 24% от общего количества. Фича на удаление.

1.27. Информация event_value содержит только пропуски. Фича на удаление.